File: 2ndSemester/kursprogramowania/lista3/cpp/src/circle.cpp
#include <circle.hpp>

#include <cmath>
#include <string>

Circle::Circle(double radius) noexcept(false)
{
    if (radius <= 0)
        throw std::invalid_argument(std::string(__FILE__) + ":" + std::to_string(__LINE__) + " [" + __func__ + "] " + "radius should be positive, got " + std::to_string(radius));

    this->radius = radius;
}

double Circle::area() const noexcept(true) 
{
    return M_PI * this->radius * this->radius;
}

double Circle::perimeter() const noexcept(true)
{
    return 2 * M_PI * this->radius;
}

const std::string Circle::name() const noexcept(true) 
{
    return "Circle";
}

File: 2ndSemester/kursprogramowania/lista3/java/Circle.java
/**
 * Circle
 */
public final class Circle extends Figure 
{
    private final double radius;
    
    public Circle(final double radius)
    {
        if (radius <= 0) 
        {
            throw new IllegalArgumentException("Radius should be positive, got " + radius);    
        }
        this.radius = radius;
    }

    @Override
    public double area() 
    {
        return Math.PI * radius * radius;
    } 

    @Override
    public double perimeter()
    {
        return 2 * Math.PI * radius;
    }

    @Override
    public String name()
    {
        return "Circle";
    }
}
